### PR TITLE
Support watch order without uuid

### PR DESCRIPTION
**Pull request by matanporeh**

### CODE REVIEW

File: sources/model/managars/GGHTTPClientManager.m [hunk 1]
```diff
@@ -632,17 +632,17 @@ - (void)getOrderByID:(NSUInteger)orderId
...
-                     extras:(NSDictionary * _Nullable)extras
+- (void)getOrderByShareUUID:(nonnull NSString *)shareUUID
+      accessControlParamKey:(nonnull NSString *)accessControlParamKey
+    accessControlParamValue:(nonnull NSString *)accessControlParamValue
+                     extras:(nullable NSDictionary *)extras
       withCompletionHandler:(nullable GGOrderResponseHandler)completionHandler {
     
     NSMutableDictionary *params = [NSMutableDictionary dictionary];
-    [self addAuthinticationToParams:&params];
+    [params setObject:accessControlParamValue forKey:accessControlParamKey];
```
Comment (orxelm) [inline sources/model/managars/GGHTTPClientManager.m]: can crash if `accessControlParamValue` is nil

File: tests/BringgTrackingClientTests.m [hunk 2]
```diff
@@ -119,16 +120,29 @@ - (void)sendWatchOrderWithOrderUUID:(nonnull NSString *)uuid
...
+        }else if ([secondAccessControlParamKey isEqualToString:PARAM_ORDER_UUID]) {
+            uuid = secondAccessControlParamValue;
+        }
+
+        if (uuid) {
+            id<OrderDelegate> existingDelegate = [self.orderDelegates objectForKey:uuid];
+            
+            if (existingDelegate) {
+                if (success) {
+                    [existingDelegate watchOrderSucceedForOrder:[self getOrderWithUUID:uuid]];
```
Comment (orxelm) [inline tests/BringgTrackingClientTests.m]: responds to selector for both

File: sources/model/managars/GGHTTPClientManager.m (hunk 1 above)
Comment (matanporeh) [inline sources/model/managars/GGHTTPClientManager.m]: fixed

File: tests/BringgTrackingClientTests.m (hunk 2 above)
Comment (matanporeh) [inline tests/BringgTrackingClientTests.m]: fixed

**Pull request merged**